packages/fast-components-react-base/src/auto-suggest/auto-suggest.tsx

```diff
@@ -87,9 +87,19 @@ class AutoSuggest extends Foundation<
     }
 
     public componentDidUpdate(prevProps: AutoSuggestProps): void {
-        const updatedMenuVisibility: boolean = this.validateMenuState(
-            this.state.isMenuOpen
-        );
+        let updatedMenuVisibility: boolean = this.state.isMenuOpen;
+
+        if (
```

Comment:
These two if statements seem needlessly complex:

The second statement will never pass unless the first one passes because `updatedMenuVisibility` is initialized to `this.state.isMenuOpen`. You should be able to do all the checking you need to in one if statement.

Reply:
yep, and the toggle function checks against previous state too.